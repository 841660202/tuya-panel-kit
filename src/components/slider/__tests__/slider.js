import React from 'react';
import {
  Image,
} from 'react-native';
import renderer from 'react-test-renderer';
import Slider from '../index';

const sliderHover = require('./res/slider_hover.png');

describe('Slider Component', () => {
  it('basic render', () => {
    const component = renderer.create(<Slider />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('onlyMaximumTrack render', () => {
    const component = renderer.create(<Slider onlyMaximumTrack={true} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('horizontal render', () => {
    const fn = value => console.log(value, '=== value ===');
    const HorizontalSlider = Slider.Horizontal;

    const component = renderer.create(
      <HorizontalSlider
        canTouchTrack={true}
        onValueChange={fn}
        value={24}
        minimumValue={0}
        maximumValue={100}
        onlyMaximumTrack={false}
        style={{
          width: 300,
          height: 48,
        }}
        trackStyle={{
          width: 300,
          height: 30,
          top: 36,
          backgroundColor: 'transparent',
          justifyContent: 'center',
          alignItems: 'center',
        }}
        renderThumb={() =>
          (<Image
            source={sliderHover}
           />)
        }
        thumbStyle={{
          width: 300,
          height: 30,
          top: 36,
          backgroundColor: 'transparent',
          justifyContent: 'center',
          alignItems: 'center',
        }}
        minimumTrackTintColor={'#4A90E2'}
        maximumTrackTintColor={'#50E3C2'}
        horizontal={true} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('vertical render', () => {
    const fn = value => console.log(value, '=== value ===');
    const VerticalSlider = Slider.Vertical;

    const component = renderer.create(
      <VerticalSlider
        canTouchTrack={true}
        onValueChange={fn}
        value={24}
        minimumValue={0}
        maximumValue={100}
        style={{
          width: 300,
          height: 48,
        }}
        trackStyle={{
          width: 300,
          height: 30,
          top: 36,
          backgroundColor: 'transparent',
          justifyContent: 'center',
          alignItems: 'center',
        }}
        renderThumb={() =>
          (<Image
            source={sliderHover}
           />)
        }
        thumbStyle={{
          width: 300,
          height: 30,
          top: 36,
          backgroundColor: 'transparent',
          justifyContent: 'center',
          alignItems: 'center',
        }}
        minimumTrackTintColor={'#4A90E2'}
        maximumTrackTintColor={'#50E3C2'}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should update', () => {
    const fn = jest.fn();
    const component = renderer.create(
      <Slider
        key="some-key"
        canTouchTrack={true}
        onValueChange={fn}
        value={0}
        minimumValue={0}
        maximumValue={100}
        minimumTrackTintColor={'#4A90E2'}
        maximumTrackTintColor={'#50E3C2'}
      />
    );
    component.update(
      <Slider
        key="some-key"
        style={{
          width: 300,
          height: 48,
        }}
        trackStyle={{
          width: 300,
          height: 30,
          top: 36,
          backgroundColor: 'transparent',
          justifyContent: 'center',
          alignItems: 'center',
        }}
        canTouchTrack={true}
        onValueChange={fn}
        value={50}
        minimumValue={0}
        maximumValue={100}
        disabled={true}
        minimumTrackTintColor={'#4A90E2'}
        maximumTrackTintColor={'#50E3C2'}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should gesture respond properly', () => {
    const SLIDER_WIDTH = 300;
    const THUMB_SIZE = 24;
    const onSlidingStart = jest.fn();
    const onValueChange = jest.fn();
    const onSlidingComplete = jest.fn();
    const component = renderer.create(
      <Slider
        key="some-key"
        style={{
          width: SLIDER_WIDTH,
          height: 48,
        }}
        thumbStyle={{
          width: THUMB_SIZE,
          height: THUMB_SIZE,
          borderRadius: THUMB_SIZE / 2,
        }}
        canTouchTrack={true}
        onSlidingStart={onSlidingStart}
        onValueChange={onValueChange}
        onSlidingComplete={onSlidingComplete}
        value={0}
        minimumValue={0}
        maximumValue={100}
        minimumTrackTintColor={'#4A90E2'}
        maximumTrackTintColor={'#50E3C2'}
      />
    );
    const instance = component.root.instance;
    instance._handlePanResponderGrant({ nativeEvent: { locationX: 50, locationY: 100 } });
    expect(onSlidingStart).toHaveBeenCalled();

    instance._handlePanResponderMove({ nativeEvent: { locationX: 50, locationY: 100 } }, { dx: 50, dy: 0 });
    expect(onValueChange).toHaveBeenCalled();

    instance._handlePanResponderEnd({ nativeEvent: { locationX: 50, locationY: 100 } }, { dx: 60, dy: 0 });
    expect(onSlidingComplete).toHaveBeenCalled();
  })
});
