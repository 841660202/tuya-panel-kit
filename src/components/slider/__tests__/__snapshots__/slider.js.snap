// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Component basic render 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "column",
        "height": 40,
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#b3b3b3",
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 2,
          "height": 4,
        },
        undefined,
      ]
    }
  />
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#3f3f3f",
        "borderRadius": 2,
        "height": 4,
        "opacity": 0,
        "overflow": "hidden",
        "position": "absolute",
      }
    }
  />
  <View
    collapsable={undefined}
    onLayout={[Function]}
    renderToHardwareTextureAndroid={true}
    style={
      Object {
        "backgroundColor": "#343434",
        "borderRadius": 10,
        "elevation": 2,
        "height": 20,
        "opacity": 0,
        "position": "absolute",
        "shadowColor": "rgba(0,0,0,0.3)",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": 20,
      }
    }
  />
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`Slider Component horizontal render 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "column",
        "height": 40,
      },
      Object {
        "height": 48,
        "width": 300,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#50E3C2",
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 2,
          "height": 4,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "height": 30,
          "justifyContent": "center",
          "top": 36,
          "width": 300,
        },
      ]
    }
  />
  <View
    collapsable={undefined}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 2,
        "height": 30,
        "justifyContent": "center",
        "opacity": 0,
        "overflow": "hidden",
        "position": "absolute",
        "top": 36,
        "width": 300,
      }
    }
  />
  <View
    collapsable={undefined}
    onLayout={[Function]}
    renderToHardwareTextureAndroid={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 10,
        "elevation": 2,
        "height": 30,
        "justifyContent": "center",
        "opacity": 0,
        "position": "absolute",
        "shadowColor": "rgba(0,0,0,0.3)",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "top": 36,
        "width": 300,
      }
    }
  >
    <Image
      source={1}
    />
  </View>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`Slider Component onlyMaximumTrack render 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "column",
        "height": 40,
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#b3b3b3",
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 2,
          "height": 4,
        },
        undefined,
      ]
    }
  />
  <View
    collapsable={undefined}
    onLayout={[Function]}
    renderToHardwareTextureAndroid={true}
    style={
      Object {
        "backgroundColor": "#343434",
        "borderRadius": 10,
        "elevation": 2,
        "height": 20,
        "opacity": 0,
        "position": "absolute",
        "shadowColor": "rgba(0,0,0,0.3)",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": 20,
      }
    }
  />
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`Slider Component should update 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "column",
        "height": 40,
      },
      Object {
        "height": 48,
        "width": 300,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#50E3C2",
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 2,
          "height": 4,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "height": 30,
          "justifyContent": "center",
          "top": 36,
          "width": 300,
        },
      ]
    }
  />
  <View
    collapsable={undefined}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 2,
        "height": 30,
        "justifyContent": "center",
        "opacity": 0,
        "overflow": "hidden",
        "position": "absolute",
        "top": 36,
        "width": 300,
      }
    }
  />
  <View
    collapsable={undefined}
    onLayout={[Function]}
    renderToHardwareTextureAndroid={true}
    style={
      Object {
        "backgroundColor": "#343434",
        "borderRadius": 10,
        "elevation": 2,
        "height": 20,
        "opacity": 0,
        "position": "absolute",
        "shadowColor": "rgba(0,0,0,0.3)",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": 20,
      }
    }
  />
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`Slider Component vertical render 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "row",
        "width": 40,
      },
      Object {
        "height": 48,
        "width": 300,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#50E3C2",
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 2,
          "width": 4,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "height": 30,
          "justifyContent": "center",
          "top": 36,
          "width": 300,
        },
      ]
    }
  />
  <View
    collapsable={undefined}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 2,
        "height": 30,
        "justifyContent": "center",
        "opacity": 0,
        "overflow": "hidden",
        "position": "absolute",
        "top": 36,
        "width": 300,
      }
    }
  />
  <View
    collapsable={undefined}
    onLayout={[Function]}
    renderToHardwareTextureAndroid={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 10,
        "elevation": 2,
        "height": 30,
        "justifyContent": "center",
        "opacity": 0,
        "position": "absolute",
        "shadowColor": "rgba(0,0,0,0.3)",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "top": 36,
        "width": 300,
      }
    }
  >
    <Image
      source={1}
    />
  </View>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />
</View>
`;
